python/ray/data/block.py

```diff
@@ -206,6 +207,8 @@ class BlockMetadata:
     input_files: Optional[List[str]]
     #: Execution stats for this block.
     exec_stats: Optional[BlockExecStats]
+    #: Node where this block was created.
+    location: Optional[NodeIdStr]
```

Comment:
any reason why this needs to be optional?

Reply:
Ah because the location isn't known before the task starts to execute.

Reply:
I mean, don't we only create the metadata objects when the blocks are generated? Is there any place where we create dummy metadata objects without blocks?